### ISSUE TITLE
Happymh's search function is broken

### ISSUE THREAD

**Issue by LordYanglong** [keiyoushi/tachiyomi-extensions-source#4158]
### Source information

Happymh 1.4.11

### Source language

Chinese

### Steps to reproduce

1.Using the search function, after loading for a while, it shows that Cloudflare cannot be bypassed. 
2.I've tried changing many nodes and user agents, but the issue remains unresolved.

### Expected behavior

The search function within the plugin is not affected by Cloudflare's blocking.

### Actual behavior

Previously, there was a period where the plugin's search function was being blocked by Cloudflare. This issue was resolved after the last update. However, as of this afternoon, the plugin's search function is once again being blocked by Cloudflare.

### Mihon/Tachiyomi version

0.16.5

### Android version

Android 12

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

**Comment by Smol-Ame** [comment]
The extension/search works fine on my end.

After changing user-agents, remember to restart the app for it to take effect. Also, the extension itself seems to have its own user-agent setting you might need to tinker with (Mihon Browse Page > Extensions tab > HappyMH Settings). After changing the user-agent, see if it passes in WebView, maybe click around the site to see if any other verification shows up.

![Screenshot_20240722_075950_Mihon](https://github.com/user-attachments/assets/0722611c-f957-4490-bafc-85c6523535c5)
